assayist/processor/container_go_analyzer.py

```diff
@@ -0,0 +1,147 @@
+# SPDX-License-Identifier: GPL-3.0+
+
+import os
+import subprocess
+
+from assayist.common.models import content, source
+from assayist.processor.base import Analyzer
+from assayist.processor.utils import assert_command
+from assayist.processor.logging import log
+
+
+class ContainerGoAnalyzer(Analyzer):
+    """Analyze the Go executables in a container image layer."""
+
+    # goversion executable path.
+    GOVERSION = 'goversion'
+
+    # Backvendor executable path.
+    BACKVENDOR = 'backvendor'
+
+    # Output format template for backvendor.
+    # The module name will be followed by fields in this format.
+    BACKVENDOR_TEMPLATE = '\t{{.Ver}}\t{{.Repo}}\t{{.Rev}}'
+
+    # Types of identified source modules
+    UPSTREAM = 'upstream'
+    VENDORED = 'embedded_source_locations'
+
+    def run(self):
+        """Start the container Go analyzer."""
+
+        # Check we have access to the executables we need.
+        assert_command(self.BACKVENDOR)
+        assert_command(self.GOVERSION)
+
+        build_info = self.read_metadata_file(self.BUILD_FILE)
+        build_id = build_info['id']
+        if not self.is_container_build(build_info):
+            log.info(f'Skipping build {build_id} because the build is not a container')
+            return
+
+        # This container's build is assumed to exist since it is
+        # created by the main analyzer.
+        build = content.Build.nodes.get(id_=build_id)
+        source_location = build.source_location
+        if not source_location:
+            log.error('Missing source location for container build')
+            return
+
+        self._process_source_code(source_location)
+        self._claim_go_executables()
+
+    def _process_source_code(self, source_location):
+        """Run backvendor on the source code.
+
+        :param SourceLocation source_location: local source code DB node
+        """
+        # Find the source code.
+        srcdir = os.path.join(self.input_dir, self.SOURCE_DIR)
```

Comment:
`self.input_dir` defaults to `/metadata` so this will construct a path of `/metadata/source` whereas we probably want `/source`.

Reply:
Maybe `Analyzer.SOURCE_DIR` should be "../source" then?

Reply:
Or we could point the input directory to the parent of `source/` and `metadata/` and store the individual directory names as constants in `Analyzer`. That way, you could join `self.input_dir` and `self.XYZ_DIR` to always get the desired path. So far, we have four directories: metadata, source, unpacked_archives, and output_files.

Reply:
Added a commit to do this.